src/util/antoraSupport.ts

```diff
@@ -0,0 +1,36 @@
+import vscode from 'vscode'
+import fs from 'fs'
+import yaml from 'js-yaml'
+import path from 'path'
+
+export default class AntoraSupport {
+  public async isDetected (): Promise<boolean> {
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 10)
+    return antoraYmlFiles.length > 0
+  }
+
+  public async getAntoraYmlPath (textDocument: vscode.TextDocument): Promise<any> {
+    const pathToAsciidocFile = textDocument.uri.fsPath
+    const antoraYmlFiles = await vscode.workspace.findFiles('**/antora.yml', '/node_modules/', 100)
+    // check if antora.yml file matches with textDocument content source root
+    try {
+      for (const antoraYmlFile of antoraYmlFiles) {
+        if (pathToAsciidocFile.startsWith(path.dirname(antoraYmlFile.path))) {
```

Comment:
we should make sure that the `pathToAsciidocFile` starts with `/path/to/workspace/repository/modules/<module>/pages`.


```js
// pseudo-code (untested)
const asciidocFilename = path.basename(pathToAsciidocFile)
const modulesPath = path.join(path.dirname(antoraYmlFile.path), 'modules')
if (pathToAsciidocFile.startsWith(modulesPath) && pathToAsciidocFile.slice(modulesPath.length).match(/^\/[^/]+\/pages\/.*/)) {
  return antoraYmlFile.path
}
```

Reply:
you can add the following test case:

```
/path/to/workspace/repository/antora.yml
/path/to/workspace/repository/README.adoc
```

`README.adoc` should not have Antora support enable (because it does not belong to an Antora module)